### PR TITLE
add action show-config

### PR DESCRIPTION
**Pull request by sed-i**

### CODE REVIEW

**Review by mmanciop — APPROVED** [review]

File: actions.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+show-config:
+  description: Show alertmanager config file.
+show-silences:
```
Comment (balbirthomas) [inline actions.yaml:3]: English NitPick: "show-silences" or "show-silenced" ? Same about variable names below.

**Review by balbirthomas — APPROVED** [review]

File: actions.yaml (hunk 1 above)
Comment (sed-i) [reply actions.yaml:3]: this will show the "silences" not the silenced alarms. AM GUI has a tab named "silences" too

File: src/charm.py [hunk 2]
```diff
@@ -46,18 +46,34 @@ def reload(self) -> bool:
...
 
-    def status(self) -> Optional[dict]:
-        url = urllib.parse.urljoin(self.base_url, "/api/v2/status")
+    @staticmethod
+    def _get(url: str) -> Optional[dict]:
         try:
             response = requests.get(url)
             if response.ok and response.status_code == 200:
-                status = json.loads(response.text)
+                text = json.loads(response.text)
```
Comment (Abuelodelanada) [inline src/charm.py:54]: I think that checking both `response.ok` and `response.status_code == 200` [is redundant](https://github.com/psf/requests/blob/8149e9fe54c36951290f198e90d83c8a0498289c/requests/models.py#L696)

File: src/charm.py [hunk 3]
```diff
@@ -46,18 +46,34 @@ def reload(self) -> bool:
...
+        return text
+
+    def status(self) -> Optional[dict]:
+        url = urllib.parse.urljoin(self.base_url, "/api/v2/status")
+        return self._get(url)
+
+    def silences(self, state: str = None) -> Optional[List[dict]]:
+        url = urllib.parse.urljoin(self.base_url, "/api/v2/silences")
+        silences = self._get(url)
+        logger.info("silences (all): %s", silences)
```
Comment (Abuelodelanada) [inline src/charm.py:69]: Shouldn't be `logger.debug` ?

**Review by Abuelodelanada — APPROVED** [review]

File: src/charm.py (hunk 2 above)
Comment (sed-i) [reply src/charm.py]: removed "ok"

File: src/charm.py (hunk 3 above)
Comment (sed-i) [reply src/charm.py]: removed

**Pull request merged**